### ISSUE TITLE
brew binary distributions are not stable (LLVM 3.9.1 breaks Mull)

### ISSUE THREAD

**Issue by stanislaw** [mull-project/mull#180]
Currently `brew install llvm@3.9` installs LLVM 3.9.1 which causes Mull to break somewhere around the code:

```
cl::OptionCategory MullOptionCategory("Mull");

static cl::opt<std::string> ConfigFile(
    llvm::cl::desc("<config file>"),
    llvm::cl::Positional
);
```

which doesn't parse `config.yml` argument anymore.

We need to find a stable way of always getting the right version of LLVM's binaries.

**Comment by stanislaw** [comment]
Ok, whatever the problem with `cl::*` guys is, I think we need to switch to getting the binaries from: [LLVM Download Page](http://releases.llvm.org/) and never from `brew` anymore.

For that we finally have to teach CMake to just work with `llvm-config`:

```bash
$ llvm-config --libs x86
-lLLVMX86Disassembler -lLLVMX86AsmParser -lLLVMX86CodeGen -lLLVMSelectionDAG -lLLVMAsmPrinter -lLLVMDebugInfoCodeView -lLLVMCodeGen -lLLVMTarget -lLLVMScalarOpts -lLLVMInstCombine -lLLVMInstrumentation -lLLVMTransformUtils -lLLVMBitWriter -lLLVMAnalysis -lLLVMProfileData -lLLVMX86Desc -lLLVMObject -lLLVMMCParser -lLLVMBitReader -lLLVMMCDisassembler -lLLVMX86Info -lLLVMX86AsmPrinter -lLLVMMC -lLLVMX86Utils -lLLVMCore -lLLVMSupport
```

I can work on this first thing.

**Comment by stanislaw** [comment]
Closed by #187 .
